model/generate.go

```diff
@@ -312,6 +299,15 @@ func (g *GeneratedProject) saveNewBuildsAndTasks(ctx context.Context, v *Version
 	return nil
 }
 
+// SetNewTasksAndActivationInfo sets the NewTVPairs and ActivationInfo fields on the generator.
+func (g *GeneratedProject) SetNewTasksAndActivationInfo(ctx context.Context, v *Version, p *Project) {
```

Comment:
One suggestion on how to write this a little more cleanly is to change this to a `GetNewTasks` function, then check if `g.NewTVPairs` is initialized, and just return it if so. If not, then initialize it once. Something like:
```go
if g.NewTVPairs != nil {
    // Already initialized
    return g.NewTVPairs
}

// First initialization
g.NewTVPairs = g.getNewTasks...
return g.NewTVPairs
// And then make some equivalent function for activation info as well
```
It would also get around having explicitly call `Set` to ensure the data is set before we actually need the fields.

Reply:
Sounds good, changed!